### DISCUSSION TITLE
Alerts from apparmor about telegraf in docker

### DISCUSSION THREAD

**Discussion by DebianGuru** [Security-Onion-Solutions/securityonion#2860] [Unsupported Versions] [answered]
Greetings,

I get alerts every 30 seconds or so from ossec module that is also shown in the syslog.  The error is:  

type=1400 audit(1612440750.062:2391): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=7096 comm="telegraf" requested_mask="read" denied_mask="read" peer="unconfined"

This is causing thousands of alerts per day.

I have googled this to death.  I don't really have any experience with docker or apparmor and have a minimal understanding.  I've read some similar posts that all point back to some "docker-default" profile for docker.  If I understand, apparmor has a config for each docker container (telegraf in this case) that sets access controls.  It appears that the "docker-default" file is forbidding this particular operation, thus causing the alerts.

So I think I understand how this works, but I'm not sure how to modify the "docker-default" file within the telegraf container to prevent this alert OR is this legitimate and I need to find out how to prevent telegraf from misbehaving.

Any help is appreciated.  I'm feeling it out before I replace our old SO setup in a few weeks.

Here's the details:

This is a standalone SO 2.3.21 system
Net installed on Ubuntu  18.04.5 LTS
so-status shows docker and all container status as "OK"
salt-call state.highstate returns: 
     local:
          Data failed to compile:
          ----------
          The function "state.highstate" is running as PID 6262 and was started at  with jid req

Any help is appreciated.

**Comment by weslambert** [comment, marked as answer]
We are tracking this here: https://github.com/Security-Onion-Solutions/securityonion/issues/2560.  In the meantime, you could use custom Wazuh rules to override the behavior: https://docs.securityonion.net/en/latest/wazuh.html?highlight=wazuh#tuning-rules

**Comment by DebianGuru** [comment]
Thanks for the reply, but that literally just says, "You can add new rules and modify existing rules in /opt/so/rules/hids/local_rules.xml."  There's no indication of how that is done.  I googled some Wazuh documentation that talked about adding or modifying rules.  Nothing clear cut about how to disable or ignore a rule.  At least, nothing I could find.  As a matter of fact, I'm finding it difficult to get information on how to enable/disable any wazuh rules.  It seems by default that EVERY possible notification is turned on and I'm being buried in alerts from every little thing.  Do you have any idea where I can find decent documentation or tutorial on this?  Thanks.

**Reply by launchdaemon** [reply]
Hi @DebianGuru 
I was having the same problem as you.  After reading the Wazuh [documentation](https://documentation.wazuh.com/4.0/user-manual/ruleset/custom.html#changing-an-existing-rule) I modified the rule to exclude any hits with a mention of `telegraf` (rather than disable the rule all together).

I did this by copying the original rule from `/opt/so/rules/hids/ruleset/rules/0315-apparmor_rules.xml` and adding it to `/opt/so/rules/hids/local_rules.xml`.  There is actually an open issue on the Wazuh repo for allowing multiple `match` statements and negating them although I can't find it now.  This would have been perfect but doesn't appear to have been implemented yet.  I therefore used a regex to exclude any `apparmor` log entry that also included `telegraf` which resulted in this:

```
    <group name="local,syslog,apparmor,">
    
      <rule id="52002" level="3" overwrite="yes">
        <if_sid>52000</if_sid>
        <status>DENIED</status>
        <!-- <match> apparmor=</match> --><!-- This was the previous check. -->
        <regex>^(?!.*(telegraf)).*apparmor.*$</regex>
        <description>Apparmor DENIED</description>
      </rule>
    
    </group> 
```
I am definitely not an expert though so would encourage you to read the Wazuh documentation as well as it was very helpful, also to be sure that above is suitable for you environment.

**Reply by DebianGuru** [reply]
Thanks for the answer!  I was just going to set priority to 0 to shut it up, but this is much better. But when I tried it I got an error in the regex.  Are you sure that was your string you entered?
